core/src/main/java/org/elasticsearch/transport/RequestHandlerRegistry.java

```diff
@@ -52,8 +55,14 @@ public Request newRequest() {
             return requestFactory.get();
     }
 
-    public TransportRequestHandler<Request> getHandler() {
-        return handler;
+    public void processMessageReceived(Request request, TransportChannel channel) throws Exception {
+        final Task task = taskManager.register(channel.channelType(), action, request);
```

Comment:
so we are creating a task for every request? is this intended or should we ask the handler if it's needed?